### ISSUE TITLE
Support MAPE as built-in eval_metric.

### ISSUE THREAD

**Issue by brightcoder01** [dmlc/xgboost#5965]
[MAPE](https://en.wikipedia.org/wiki/Mean_absolute_percentage_error) is a very useful to measure the accuracy of a forecasting method and the loss of regression model.
From the eval_metrics list in [the API doc](https://xgboost.readthedocs.io/en/latest/parameter.html#learning-task-parameters), we have [MAE](https://en.wikipedia.org/wiki/Mean_absolute_error) but don't have MAPE. In many scenarios, MAPE can provide more insight to the performance of the model.
It will be useful to add this metrics as the built-in eval_metrics when we are training a regression model using XGBoost.

**Comment by brightcoder01** [comment]
Gentle ping. @trivialfis How do you think of this feature request?

**Comment by trivialfis** [comment]
I will look into it.  Should be easy enough.  Currently I'm cleaning up the old code base for categorical data support so it might take a while.  Would you like to open a PR? I'm happy to help.

**Comment by hcho3** [comment]
@trivialfis I can take a shot.

**Comment by trivialfis** [comment]
Thanks!

**Comment by hcho3** [comment]
#6119 adds MAPE.
